oidc-wp/openid-connect-generic
# added define config with OIDC_LOGIN_TYPE and OIDC_CLIENT_SCOPE

### All Submissions:


### Changes proposed in this Pull Request:

I add here support for define() settings for two variables: `login_type` and `scope`.

`OIDC_CLIENT_SCOPE` is a space-separated list of scopes.
`OIDC_LOGIN_TYPE` can either be `button` or `auto`. Other values have undefined behaviour.
`OIDC_ENFORCE_PRIVACY` (boolean): enable/disable the eponymous option
`OIDC_CREATE_IF_DOES_NOT_EXIST` (boolean) enable/disable the eponymous option
`OIDC_LINK_EXISTING_USERS` (boolean) enable/disable the eponymous option
`OIDC_REDIRECT_USER_BACK` (boolean) enable/disable the eponymous option
`OIDC_REDIRECT_ON_LOGOUT` (boolean) enable/disable the eponymous option

### How to test the changes in this Pull Request:

1. set `define('OIDC_CLIENT_SCOPE', 'email openid profile')` in your wp-config.php
2. see it shows up in the config and the variable is read-only
3. do the same for `OIDC_LOGIN_TYPE` and the other variables.

### Other information:




### Changelog entry

Added seven new environment variables matching the options in the settings page.
Setting these `define`s disables changing the options in the interface.

`OIDC_CLIENT_SCOPE`: a space-separated list of scopes.
`OIDC_LOGIN_TYPE`: (`button`|`auto`)
`OIDC_ENFORCE_PRIVACY` (boolean): enable/disable the eponymous option
`OIDC_CREATE_IF_DOES_NOT_EXIST` (boolean) enable/disable the eponymous option
`OIDC_LINK_EXISTING_USERS` (boolean) enable/disable the eponymous option
`OIDC_REDIRECT_USER_BACK` (boolean) enable/disable the eponymous option
`OIDC_REDIRECT_ON_LOGOUT` (boolean) enable/disable the eponymous option
